### DISCUSSION TITLE
I have a few questions about lists and their combinations

### DISCUSSION THREAD

**Discussion by Psentinel** [adieyal/sd-dynamic-prompts#367] [Q&A] [answered]
I have a long list of creatures.

 I want to combine them without spaces or commas or anything.
 
  For example - 
  
  my list has dragon, and dog in it. 
  
  I want my prompt to read: "dragondog" 
  
   How do I accomplish this? 
   
    I also want to have my first list pick 1 or 2 options, but with your current syntax, I can only pick ONE + one random one. But the first one is locked in. 
    
     I want 1 or 2 completely random choices.
     
      Is there a way to accomplish this?

**Comment by adieyal** [comment, marked as answer]
Typically, you would do something like

`{1-2$$dog|dragon|cow|eagle}`

That will give you,
```
dog
cow,dragon
eagle,dragon
...
```

if you want to change the separator from the default `,` you can do this:

`{1-2$$-$$dog|dragon|cow|eagle}`

which will produce:
```
dog
cow-dragon
eagle-dragon
```

You don't want a space between them though, and this doesn't work:
`{1-2$$$$dog|dragon|cow|eagle}`

A workaround might be:
```
{0-1$$dog|dragon|cow|eagle}{dog|dragon|cow|eagle}
```

or perhaps this would be neater.
```
{0-1$$__creature__}__creature__
```

You could also try Jinja templates, but the result is a little gross:
```jinja
{% for i in range(10) %}
    {% prompt %}
        {{ random_sample("{1-2$$dog|dragon|cow|eagle}") | replace(",", "") }}
    {% endprompt %}
{% endfor %}
```

The will generate the combinations with a `,` and will then replace it with an empty string.

**Reply by Psentinel** [reply]
Thanks man! That was a huge help. Still learning my way around these kind of things.

 : ) <3

**Reply by adieyal** [reply]
Great. Glad it works for you

**Comment by manchuwook** [comment]
Dumb question: If yaml files end in .yaml, and json files end in .json - what is the extension for Jinja2 files? .jinja2?  Or does it have to be pasted in the prompt box?
